pkg/compiler/types.go

```diff
@@ -793,6 +803,29 @@ var typeArgSizeRange = &typeArg{
 	},
 }
 
+var typeArgSizeRangeLiteral = &typeArg{
+	AllowColon: true,
+	CheckConsts: func(comp *compiler, t *ast.Type) {
+		if t.Ident == "len" {
+			return
+		}
+		if !t.HasColon {
+			t.Value2 = t.Value
+		}
+		const maxVal = 1e6
+		if t.Value > t.Value2 || t.Value > maxVal || t.Value2 > maxVal {
+			comp.error(t.Pos, "bad size range [%v:%v]", t.Value, t.Value2)
+		}
+	},
+	Check: func(comp *compiler, t *ast.Type) {
+		if t.Ident == "len" {
+			if len(t.Args) != 1 {
```

Comment:
We also need to check that when it's not "len" number of args is equal to 0.

On the second though, it's actually an oversight that we allow subargs for args at all.
I've fixed one case that violated this:
https://github.com/google/syzkaller/commit/1be453f07bb52417650d9d617d41a834ecf1a0ab
and added the check:
https://github.com/google/syzkaller/commit/7a88b14122b5b70ed04bcd75932f411256e1662b

Please rebase and set MaxArgs=1 for this arg. But you still need to check that it's 0 for non-len case.

Reply:
Both done.